code/datums/components/wet_floor.dm

```diff
@@ -100,6 +103,9 @@
 	SIGNAL_HANDLER
 
 	for(var/i in time_left_list)
+		if(text2num(i) == TURF_WET_MAGICAL) // magical lube disappears regardless of given `duration_decrease` value
+			time_left_list[i] -= 1 SECONDS
+			continue
```

Comment:
This logic seems incredibly strange and makes no sense to me. Is dry() only ever called as part of process? Perhaps you should just use a timer instead to force try it and set the dry time to infinity.

Reply:
the dry code is just stupid. I can't really tell you how it is...
Let's say you have a tile with all types of lubes, then we have...
1: Water
2: Lube
4: Ice
8: Perma ice
16: Super lube
32: Magic lube
and `dry(datum/source, strength = TURF_WET_WATER, immediate = FALSE, duration_decrease = INFINITY)` is called from `/wet_floor/process()` with parameter values `dry(null, ALL, FALSE, decrease)`
`ALL` means it will you'll catch all bitflags
so, `if(text2num(i) <= strength)` is something stupid because if you give `strength` as `ice`, it will remove `water to ice`.
so, it is actually something to be changed that it only should target the given flag... the ice case I said should get only `ice`, not `water to ice`, but It's out of the scope for now, I guess.

Reply:
I think I will make a PR for dry code refactor before this PR is merged. hang on.